### PR TITLE
Implement get_files for Pagure

### LINKED ISSUE #310: Support get_files()  in Pagure
Implement `get_files()` method in Pagure.

- Structure can be seen in the docstring of the code that implements that: https://pagure.io/pagure/blob/dd207ee/f/pagure/api/project.py#_709
- Example: https://pagure.io/api/0/pagure/tree

### PR DESCRIPTION
**Pull request by mfocko**
Fixes #310

Signed-off-by: Matej Focko <mfocko@redhat.com>

### CODE REVIEW

File: ogr/services/pagure/project.py [hunk 1]
```diff
@@ -545,3 +545,31 @@ def full_repo_name(self) -> str:
...
+
+            for file in response["content"]:
+                if file["type"] == "file":
+                    yield file["path"]
+                elif recursive and file["type"] == "folder":
+                    subfolders.append(file["path"])
+
+    def get_files(
+        self, ref: str = None, filter_regex: str = None, recursive: bool = False
+    ) -> List[str]:
```
Comment (TomasTomecek) [inline ogr/services/pagure/project.py:569]: docstring here would be nice

**Review by TomasTomecek — APPROVED** [review]
LGTM

File: ogr/services/pagure/project.py (hunk 1 above)
Comment (mfocko) [reply ogr/services/pagure/project.py:569]: see #533, should be inherited from interface :)

**Review by lachmanfrantisek — APPROVED** [review]
Thanks a lot!

**Comment by mfocko** [comment]
recheck

**Pull request merged**